components/forms/src/api.rs

```diff
@@ -0,0 +1,553 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::error::*;
+use crate::schema::COMMON_COLS;
+
+use rusqlite::{Connection, NO_PARAMS};
+use serde::Serialize;
+use serde_derive::*;
+use std::time;
+use sync_guid::Guid;
+
+#[derive(Debug, Clone, Hash, PartialEq, Serialize, Deserialize, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct NewAddressFields {
+    pub given_name: String,
+
+    pub additional_name: String,
+
+    pub family_name: String,
+
+    #[serde(default)]
+    pub organization: String,
+
+    #[serde(default)]
+    pub street_address: String,
+
+    pub address_level3: String,
+
+    pub address_level2: String,
+
+    pub address_level1: String,
+
+    pub postal_code: String,
+
+    pub country: String,
+
+    #[serde(default)]
+    pub tel: String,
+
+    #[serde(default)]
+    pub email: String,
+}
+
+#[derive(Debug, Clone, Hash, PartialEq, Serialize, Deserialize, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct Address {
+    #[serde(rename = "id")]
+    pub guid: Guid,
+
+    pub address_fields: NewAddressFields,
```

Comment:
Nit: Since the type is already called `Address`, how do you feel about `fields` instead of `address_fields`? Writing out, say, `address.address_fields.tel` feels a little redundant.